gamla/data.py

```diff
@@ -56,6 +58,48 @@ def tuple_of_tuples_to_csv(
     )
 
 
+_field_getters = toolz.compose_left(
+    dataclasses.fields,
+    curried.map(toolz.compose_left(lambda f: f.name, operator.attrgetter)),
+)
+
+
+def match(dataclass_pattern):
+    """ creates a function that returns true if input matches dataclass_pattern.
+     Use typing.Any as wildcard for field value.
+     Supports recursive patterns.
+     """
+    # pattern -> ( (getter,...), pattern) -> ((getter,...), (value,...)) ->
+    # ((getter,...), (eq(value),...)) -> alljuxt( compose_left(getter,eq(value)),... )
+    return toolz.pipe(
+        dataclass_pattern,
+        toolz.juxt(toolz.compose_left(_field_getters, tuple), itertools.repeat),
```

Comment:
move the tuple to `_field_getters` and then you can lose the compose

Reply:
yeah, usually i think that these should be lazy by default and materialize only if needed (e.g. before a juxt). but that might be a premature optimization